src/cmd/services/m3coordinator/server/m3msg/types.go

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package m3msg
+
+import (
+	"sync/atomic"
+
+	"github.com/m3db/m3metrics/policy"
+	"github.com/m3db/m3msg/consumer"
+)
+
+// WriteFn is the function that writes a metric.
+type WriteFn func(
+	id []byte,
+	metricTimeNanos int64,
+	value float64,
+	sp policy.StoragePolicy,
+	callback *RefCountedCallback,
```

Comment:
Supernit: Might be nice to make this have CallbackSuccess() and CallbackFailure()

Reply:
I'm open to it, I use the current interface so it's cheaper for us to add more callback types in the future